nemo/collections/asr/models/slu_models.py

```diff
@@ -0,0 +1,325 @@
+                dim=cfg.adapter.adapter_dim,  # the bottleneck dimension of the adapter
+                activation=cfg.adapter.adapter_activation,  # activation used in bottleneck block
+                norm_position=cfg.adapter.adapter_norm_position,  # whether to use LayerNorm at the beginning or the end of the adapter
+            )
+            try:
+                self.add_adapter(name=cfg.adapter.adapter_name, cfg=adapter_cfg)
+            except ValueError:
+                logging.warning(f"Adapter name {cfg.adapter.adapter_name} already exists, skipping.")
+            self.set_enabled_adapters(name=cfg.adapter.adapter_name, enabled=True)
+            self.encoder.freeze()
+            self.unfreeze_enabled_adapters()
+
+        self.vocabulary = self.tokenizer.tokenizer.get_vocab()
+        vocab_size = len(self.vocabulary)
+
+        # Create embedding layer
+        self.cfg.embedding["vocab_size"] = vocab_size
+        self.embedding = Serialization.from_config_dict(self.cfg.embedding)
+
+        # Create token classifier
+        self.cfg.classifier["num_classes"] = vocab_size
+        self.classifier = Serialization.from_config_dict(self.cfg.classifier)
+
+        self.loss = SmoothedNLLLoss(label_smoothing=self.cfg.get("loss.label_smoothing", 0.0))
+
+        self.searcher = SequenceGenerator(
+            cfg=cfg.searcher,
+            embedding=self.embedding,
+            decoder=self.decoder,
+            log_softmax=self.classifier,
+            tokenizer=self.tokenizer,
+        )
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        if hasattr(self.preprocessor, '_sample_rate'):
+            input_signal_eltype = AudioSignal(freq=self.preprocessor._sample_rate)
+        else:
+            input_signal_eltype = AudioSignal()
+        return {
+            "input_signal": NeuralType(('B', 'T'), input_signal_eltype, optional=True),
+            "input_signal_length": NeuralType(tuple('B'), LengthsType(), optional=True),
+            "target_semantics": NeuralType(('B', 'T'), input_signal_eltype, optional=True),
+            "target_semantics_length": NeuralType(tuple('B'), LengthsType(), optional=True),
+            "processed_signal": NeuralType(('B', 'D', 'T'), SpectrogramType(), optional=True),
+            "processed_signal_length": NeuralType(tuple('B'), LengthsType(), optional=True),
+            "sample_id": NeuralType(tuple('B'), LengthsType(), optional=True),
+        }
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "log_probs": NeuralType(('B', 'T', 'D'), LogprobsType(), optional=True),
+            "lengths": NeuralType(tuple('B'), LengthsType(), optional=True),
+            "greedy_predictions": NeuralType(('B', 'T'), LabelsType(), optional=True),
+        }
+
+    def set_decoding_strategy(self, cfg: SearcherConfig):
+        max_len = getattr(self.searcher, "generator.max_seq_length", cfg.max_sequence_length)
```

Comment:
"search" and "searcher" are not good names.. hmm how about "decoding" similar to RNNT - for decoding strategy?

Reply:
Also avoid using default values for getattr, causes silent bugs

Reply:
I changed the name to `sequence_generator`, since there is another `decoding` when creating `WERBPE`